records_mover/db/mysql/load_options.py

```diff
@@ -0,0 +1,289 @@
+from sqlalchemy.sql.expression import text, TextClause
+from records_mover.records import RecordsHints
+from records_mover.utils import quiet_remove
+from records_mover.records.hints import cant_handle_hint
+from records_mover.records import DelimitedRecordsFormat
+from records_mover.records.types import (
+    HintEncoding, HintRecordTerminator,
+    HintFieldDelimiter, HintQuoteChar,
+    HintQuoting, HintEscape,
+    HintHeaderRow, HintCompression,
+    HintDoublequote,
+)
+from typing import Optional, Set, Dict, NamedTuple, NoReturn, TYPE_CHECKING
+if TYPE_CHECKING:
+    from typing_extensions import Literal
+    # http://dev.mysql.com/doc/refman/8.0/en/charset-mysql.html
+    MySqlCharacterSet = Literal['big5', 'binary', 'latin1', 'ucs2',
+                                'utf8', 'utf8mb4', 'utf16', 'utf16le',
+                                'utf32']
+
+    # The NoReturn annotations are the mypy way of validating we're
+    # covering all officially approved cases of an the hint at
+    # typechecking time.
+    #
+    # https://github.com/python/mypy/issues/6366#issuecomment-560369716
+    def invalid_hint(fail_if_cant_handle_hint: bool,
+                     hint_name: str,
+                     hints: RecordsHints,
+                     the_bad_hint: NoReturn) -> NoReturn:
+        assert False
+
+else:
+    MySqlCharacterSet = str
+
+    # The cant_handle_hint at runtime is to cover cases that slip
+    # through due to our use of "type: ignore" rather than run-time
+    # validation below.
+    def invalid_hint(fail_if_cant_handle_hint: bool,
+                     hint_name: str,
+                     hints: RecordsHints,
+                     the_bad_hint: NoReturn) -> NoReturn:
+        cant_handle_hint(fail_if_cant_handle_hint, hint_name, hints)
+
+
+MYSQL_CHARACTER_SETS_FOR_LOAD: Dict[HintEncoding, MySqlCharacterSet] = {
+    "UTF8": 'utf8',
+    "UTF16": 'utf16',
+    "UTF16LE": 'utf16le',
+    # https://dev.mysql.com/doc/refman/8.0/en/charset-unicode.html
+    #
+    # The MySQL implementation of UCS-2, UTF-16, and UTF-32 stores
+    # characters in big-endian byte order and does not use a byte
+    # order mark (BOM) at the beginning of values. Other database
+    # systems might use little-endian byte order or a BOM. In such
+    # cases, conversion of values will need to be performed when
+    # transferring data between those systems and MySQL. The
+    # implementation of UTF-16LE is little-endian.
+    "UTF16BE": 'utf16',
+    # MySQL uses no BOM for UTF-8 values.
+    "LATIN1": 'latin1',
+    #  MySQL's latin1 is the same as the Windows cp1252 character set.
+    #
+    # https://dev.mysql.com/doc/refman/8.0/en/charset-we-sets.html
+    "CP1252": 'latin1',
+}
+# TODO field -> fields below
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker